src/components/bucket-utils.js

```diff
@@ -21,18 +22,20 @@ export const withRequesterPaysHandler = _.curry((handler, fn) => async (...args)
 export const requesterPaysWrapper = ({ onDismiss }) => WrappedComponent => {
   const Wrapper = forwardRef((props, ref) => {
     const [showModal, setShowModal] = useState(false)
-    return showModal ?
-      h(Modal, {
-        title: 'Cannot access data',
+
+    return Utils.cond(
+      [showModal, () => h(RequesterPaysModal, {
         onDismiss: () => onDismiss(props),
-        showCancel: false
-      }, [
-        'This data is in a requester pays bucket.'
-      ]) :
-      h(WrappedComponent, {
+        onSuccess: selectedBilling => {
+          workspaceStore.set({ userProject: selectedBilling, ...workspaceStore.get() })
```

Comment:
By storing this in the workspace data, it will get overwritten if the workspace gets reloaded, which happens after some user interactions. It might be cleaner to put this in a separate atom, and add a listener that clears it when the workspace changes.